src/app/pages/courses/components/courses-list/courses-list.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Router } from '@angular/router';
+import { Course } from '../../../../core/models/course_model';
+import { COURSES, EDIT_COURSE } from '../../../../core/constants/routes';
+
+@Component({
+    selector: 'app-courses-list',
+    templateUrl: './courses-list.component.html',
+    styleUrls: ['./courses-list.component.scss'],
+})
+export class CoursesListComponent implements OnInit {
+    public loadMore: boolean = false;
+    public preShowNumber: number = 2;
+
+    @Input() courses!: Course[];
```

Comment:
lets avoid usage of **!** - it could lead a lot of bugs in runtime
Provide default value or make it optional instead